model/syncPackage/migration/RdsMigrationService.php

```diff
@@ -0,0 +1,181 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019  (original work) Open Assessment Technologies SA;
+ *
+ * @author Oleksandr Zagovorychev <zagovorichev@gmail.com>
+ */
+
+namespace oat\taoSyncClient\model\syncPackage\migration;
+
+
+use common_Logger;
+use common_persistence_Manager;
+use common_persistence_SqlPersistence;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Schema\SchemaException;
+use Doctrine\DBAL\Types\Type;
+use oat\oatbox\service\ConfigurableService;
+
+class RdsMigrationService extends ConfigurableService implements MigrationInterface
+{
+    const TABLE_NAME = 'sync_client_migrations';
+    const OPTION_PERSISTENCE = 'persistence';
+
+    public function __construct($options = array())
+    {
+        // if initialized within other service we need to rewrite config
+        if (!array_key_exists(self::OPTION_PERSISTENCE, $options)
+            && array_key_exists(0, $options) && count($options) === 1) {
+            $options = [self::OPTION_PERSISTENCE => current($options)];
+        }
+        parent::__construct($options);
+    }
+
+    public function getMigration($migrationId = 0)
+    {
+        $query = $this->getQueryBuilder()
+            ->select('*')
+            ->from(self::TABLE_NAME)
+            ->where(self::PARAM_SYNC_ID . ' = :syncId')
+            ->setParameter('syncId', $migrationId)
+            ->orderBy(self::PARAM_CREATED_AT)
+            ->setMaxResults(1);
+
+        $result = $query->execute()->fetchAll();
+        return $result ? current($result) : [];
+    }
+
+    public function getMigrationIdByPackage($packageName)
+    {
+        $query = $this->getQueryBuilder()
+            ->select('*')
+            ->from(self::TABLE_NAME)
+            ->where(self::PARAM_PACKAGE_NAME . ' = :packageName')
+            ->setParameter('packageName', $packageName)
+            ->setMaxResults(1);
+
+        $result = $query->execute()->fetchAll();
+        return $result ? current($result)[self::PARAM_ID] : false;
+    }
+
+    /**
+     * Returns the QueryBuilder
+     *
+     * @return QueryBuilder
+     */
+    private function getQueryBuilder()
+    {
+        return $this->getPersistence()->getPlatform()->getQueryBuilder();
+    }
+
+    /**
+     * @return array
+     */
+    public function getNextMigration()
+    {
+        return $this->getMigration();
+    }
+
+    public function add($packageName)
+    {
+        $result = $this->getPersistence()->insert(self::TABLE_NAME, [
+            static::PARAM_PACKAGE_NAME => $packageName,
+            static::PARAM_UPDATED_AT   => date('Y-m-d H:i:s'),
+            static::PARAM_CREATED_AT   => date('Y-m-d H:i:s'),
+        ]);
+        return $result === 1;
+    }
+
+    public function sync($id, $syncId)
```

Comment:
Based on what does this method do it looks like it should be called something like `setSyncId`